devices.py

```diff
@@ -59,35 +61,296 @@ class Newport(EpicsMotor):
+        self.timeout = timeout
+        self.motor_timeout = motor_timeout
+
+        # How many samples are there in the N and M directions
+        self.N_dim = N_dim
+        self.M_dim = M_dim
+
+        # if the number of samples on the pallete is not given, presume full
+        # matrix allotment 
+        if samples == None:
+            self.samples = self.N_dim * self.M_dim
+        else:
+            self.samples = samples
+
+    def accept_callibration(self,N,M,start_pt,n_pt,m_pt):
+        '''
+        Notes on coordinates:
+        
+        Z points upstream along the beam, X points horizontally, Y points
+        vertically
+
+        N,M coordinates represent points on the sample palette. N roughly
+        parallels the X axis and M roughly parallels Y. 
+
+        I,J coordinates are used to reference specific samples using the N,M
+        system.
+
+        K represents samples along the snake-shaped scanning path. 
+
+
+        n : int
+            Number of samples stepped over in the N axis for calibration
+
+        m : int 
+            Number of samples stepped over in the M axis for calibration
+
+        start_pt : np.array
+            3-length iterable (x,y,z) specifying spatial coordinate of bottom
+            left (facing downstream) sample where the scan will start from 
+
+        n_pt : np.array
+            3-length iterable (x,y,z) specifying spatial coordinate of the
+            fiducail sample on the N axis
+        
+        m_pt : np.array 
+            3-length iterable (x,y,z) specifying spatial coordinate of the
+            fiducial smaple on the M ais
+        '''
+
+        # save the origin point in XYZ space
+        self.start_pt = start_pt
+
+        # Define unit vectors on the NM plane in XYZ space
+        self.N_hat = (n_pt - self.start_pt) / N
+        self.M_hat = (m_pt - self.start_pt) / M
+
+    def locate_2d(self, i, j):
+        '''
+        return XYZ coordinates of sample i,j
+
+        i : int
+
+        j : int
+        '''
+        target_pt = self.start_pt + i * self.N_hat + j * self.M_hat
+        return target_pt
+
+    def locate_1d(self, k):
+        '''
+        return NM coordinates of sample k
+
+        k : int
+        '''
+
+        # calculate the horizontal row
+        j = int(np.floor(k/self.N_dim))
+
+        # calculate the column w/o snake-wrapping
+        i = k % self.N_dim
+
+        # apply snake-wrapping to odd columns by reversing pathing order 
+        if j % 2 :
+            i = (self.N_dim - i) - 1  
+
+        return np.array([i,j])
+
+    def move_to_sample_2d(self, i ,j, timeout=None, wait=True):
+        '''
+        Move to point IJ in NM space.
+        
+        i : int
+        
+        j : int
+
+        timeout : int
+
+        wait : bool
+        '''
+        if timeout == None:
+            timeout = self.timeout
+
+        XYZ_target = self.locate_2d(i,j)
+        status = self.mv(*XYZ_target,timeout=timeout,wait=wait)
```

Comment:
PEP8 - Add spaces between arguments. Also, another way to save space is the following:

```
status = self.mv(*XYZ_target, timeout=timeout or self.timeout, wait=wait)
```